### PR TITLE
Fix lookahead of soft keywords in the PEG parser

### PR DESCRIPTION
**Pull request by pablogsal**
Automerge-Triggered-By: @gvanrossum

### CODE REVIEW

**Comment by pablogsal** [comment]
I was playing a bit with the feature and realized that we missed the lookahead code.

File: Lib/_s.py [hunk 1]
```diff
@@ -0,0 +1,499 @@
+string = str
```
Comment (gvanrossum) [inline Lib/_s.py]: What is this file?

**Comment by lysnikolaou** [comment]
When were soft keywords added? I can't remember us implementing that.

**Comment by pablogsal** [comment]
> When were soft keywords added? I can't remember us implementing that.

https://github.com/python/cpython/pull/20370

File: Lib/_s.py (hunk 1 above)
Comment (pablogsal) [inline Lib/_s.py]: This file should not be here, I will remove it

**Review by gvanrossum — COMMENTED** [review]
Okay, everything else LGTM. I now remember noticing this (during my hack to add a `print` statement) and thinking "oh, I need to add `_PyPegen_lookahead_with_string`" -- and then forgetting about it.

**Review by gvanrossum — APPROVED** [review]
Thanks!

**Comment by lysnikolaou** [comment]
> > When were soft keywords added? I can't remember us implementing that.
> 
> #20370

Nice.

File: Lib/_s.py (hunk 1 above)
Comment (pablogsal) [reply Lib/_s.py]: That was for another PR and was misadded when switching branches (https://github.com/python/cpython/pull/20436#discussion_r430751378)

**Comment by gvanrossum** [comment]
There's one caveat: we need to remind people to use `'single'` quotes for keywords, not `"double"` quotes. This happens to be the current convention but it may be surprising to people that it matters, and some folks may just automatically type `"double"` quotes.

**Comment by pablogsal** [comment]
> There's one caveat: we need to remind people to use `'single'` quotes for keywords, not `"double"` quotes. This happens to be the current convention but it may be surprising to people that it matters, and some folks may just automatically type `"double"` quotes.

That should be on the "new parser guide" on the dev guide. I was preparing something for that, but I am currently waiting to see how we want to approach https://github.com/python/cpython/pull/19969 first.

**Pull request merged**